server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -326,6 +326,11 @@ public Float parse(XContentParser parser, boolean coerce) throws IOException {
                 return parsed;
             }
 
+            @Override
+            public String familyTypeName() {
+                return NumberFieldMapper.NumberType.FLOAT.typeName();
```

Comment:
I decided to keep this very simple instead of adding more mechanism/abstraction for Type Families here.  I figure we can decide to add that once we have a few more use cases to generalize from. Happy to change that though based on feedback. WDYT?